### PR TITLE
add test for restart worker thread

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Refactor

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Format code with the latest stable rustfmt

## Overview

Add test for restart worker thread.

### CODE REVIEW

**Comment by fakeshadow** [comment]
There is some additonal finding from this test.
Like a panic dropping worker thread have a race condition where it does not prevent new connection to be accepted. 

I believe the PR should be merged first and the problems can be looked into later.

File: actix-server/src/worker.rs [hunk 1]
```diff
@@ -135,6 +135,13 @@ pub(crate) struct ServerWorker {
     shutdown_timeout: Duration,
 }
 
+/// Set worker to unavailable when dropping.
+impl Drop for ServerWorker {
+    fn drop(&mut self) {
+        self.rx.close();
```
Comment (robjtede) [inline actix-server/src/worker.rs:141]: This already happens when the receiver is dropped, no?

Comment (fakeshadow) [reply actix-server/src/worker.rs:141]: The drop order could be different than explicit call close and we want Accept to know the close asap. Like every cycle matters.

Comment (robjtede) [reply actix-server/src/worker.rs]: the drop order is well defined in rust, rx is the first field and will be dropped first

Comment (fakeshadow) [reply actix-server/src/worker.rs]: Future maintainer could be change the order and accidently make the time window larger.
And `UnboundedReciever` drop does not call close immediately. It does an atomic read first and return if there is no sender anymore. It's an optimization on the channel part but it make the close a bit later.

Comment (robjtede) [reply actix-server/src/worker.rs]: yep, if you think it's important to make this explicit then we'll keep it

Comment (fakeshadow) [reply actix-server/src/worker.rs]: I have added a note for ServerWorker field and removed the close in drop

**Review by robjtede — APPROVED** [review]

**Pull request merged**